plugins/auth-backend/src/service/router.ts

```diff
@@ -139,7 +139,16 @@ export async function createRouter({
 
   router.use('/:provider/', req => {
     const { provider } = req.params;
-    throw new NotFoundError(`No auth provider registered for '${provider}'`);
+    if (providers.includes(provider)) {
```

Comment:
This may sound weird, but instead of this kinda "side effect" behavior, how about you add a handler inside the `catch` above instead?

```ts
router.use(`/${providerId}`, () => {
  // throw an actual error that can even include info out of the exception unless that leaks info
});
```

Reply:
Good point, yeah

Reply:
Done! Updated the PR description as well. Basically removed that "catch-all" endpoint.